src/NUnitFramework/tests/Issue4584.cs

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Charlie Poole, Rob Prouse and Contributors. MIT License - see LICENSE.txt
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json.Nodes;
+
+namespace NUnit.Framework.Tests
+{
+    [TestFixture]
+    public class Issue4584
+    {
+        #region Multiline selection failure
+        [TestFixture(Description = "https://github.com/nunit/nunit/issues/4584")]
+        public class SelectionFail
+        {
+            [TestCaseSource(nameof(Data)), Explicit]
+            public void TestFailure(JsonArray jsonArray, int maxSize)
```

Comment:
I've been using this for investigation.
Eventually, if we do decide to make code changes, I will update this test to instead discover the copy that lives in `testdata` and validate it instead

Reply:
I have updated this with a test showing the filter issue and not depending on Json.